src/stable_hasher.rs

```diff
@@ -51,31 +102,61 @@ pub struct StableHasher {
 /// struct Hash128(u128);
 ///
 /// impl StableHasherResult for Hash128 {
+///     type Hash = [u64; 2];
+///
 ///     fn finish(hash: [u64; 2]) -> Hash128 {
 ///         let upper: u128 = hash[0] as u128;
 ///         let lower: u128 = hash[1] as u128;
-///    
+///
 ///         Hash128((upper << 64) | lower)
 ///     }
 /// }
 /// ```
 pub trait StableHasherResult: Sized {
+    type Hash;
+
     /// Retrieving the finalized state of the [`StableHasher`] and construct
     /// an [`Self`] containing the hash.
-    fn finish(hasher: [u64; 2]) -> Self;
+    fn finish(hash: Self::Hash) -> Self;
```

Comment:
Let's rename this method to `from` since this is supposed to be a straightforward conversion, right?

Reply:
I'm a bit sceptical of just renaming the method to `from` alone. What about also renaming the trait to `FromStableHash`?

```rust
pub trait FromStableHash: Sized {
    type Hash;

    /// Convert the finalized state of a [`StableHasher`] and construct
    /// an [`Self`] containing the transformed hash.
    fn from(hash: Self::Hash) -> Self;
}
```

Reply:
(pushed the renaming to `FromStableHash`)

Reply:
Yes, I like that 👍